From edf9b718d3007bdb519b3d64cb1f8145bba8e15e Mon Sep 17 00:00:00 2001
From: Justin Lecher <jlec@gentoo.org>
Date: Tue, 12 Feb 2013 11:55:17 +0100
Subject: [PATCH] Use pkg-config to detect json or json-c

There was a typo in the configure.ac which prevented correct usage of pkg-config
to detect json-c.
This patch corrects this typo and removes all paths from json.h in source files.
Moreover, since json-c-0.11 the pkgconfig was renamed to json-c so the
configure.ac file was modified to check for the new one if the old one was
not found.

Reported-by: Thomas D. <whissi@whissi.de>
Tested-by: Thomas D. <whissi@whissi.de>
Signed-off-by: Justin Lecher <jlec@gentoo.org>
Signed-off-by: Markos Chandras <hwoarang@gentoo.org>

---
 action.c                          | 2 +-
 configure.ac                      | 2 +-
 plugins/imkmsg/kmsg.c             | 2 +-
 plugins/mmjsonparse/mmjsonparse.c | 2 +-
 plugins/mmnormalize/mmnormalize.c | 2 +-
 plugins/ommongodb/ommongodb.c     | 2 +-
 runtime/msg.c                     | 2 +-
 runtime/msg.h                     | 2 +-
 template.c                        | 2 +-
 template.h                        | 2 +-
 10 files changed, 10 insertions(+), 10 deletions(-)

Index: rsyslog-7.2.7/action.c
===================================================================
--- rsyslog-7.2.7.orig/action.c
+++ rsyslog-7.2.7/action.c
@@ -98,7 +98,7 @@
 #include <strings.h>
 #include <time.h>
 #include <errno.h>
-#include <json/json.h>
+#include <json.h>
 
 #include "dirty.h"
 #include "template.h"
Index: rsyslog-7.2.7/configure.ac
===================================================================
--- rsyslog-7.2.7.orig/configure.ac
+++ rsyslog-7.2.7/configure.ac
@@ -34,7 +34,9 @@ PKG_PROG_PKG_CONFIG
 # modules we require
 PKG_CHECK_MODULES(LIBESTR, libestr >= 0.1.5)
 PKG_CHECK_MODULES(LIBEE, libee >= 0.4.0)
-PKG_CHECK_MODULES([JSON_C], [json])
+PKG_CHECK_MODULES([JSON_C], [json],, [
+	PKG_CHECK_MODULES([JSON_C], [json-c])
+])
 
 case "${host}" in
   *-*-linux*)
@@ -764,7 +766,7 @@ if test "x$enable_rsyslogrt" = "xyes"; t
   RSRT_LIBS1="\$(top_builddir)/runtime/librsyslog.la"
 fi
 AM_CONDITIONAL(ENABLE_RSYSLOGRT, test x$enable_rsyslogrt = xyes)
-RSRT_CFLAGS="\$(RSRT_CFLAGS1) \$(LIBESTR_CFLAGS) \$(JSON_C_FLAGS)"
+RSRT_CFLAGS="\$(RSRT_CFLAGS1) \$(LIBESTR_CFLAGS) \$(JSON_C_CFLAGS)"
 RSRT_LIBS="\$(RSRT_LIBS1) \$(LIBESTR_LIBS) \$(JSON_C_LIBS)"
 AC_SUBST(RSRT_CFLAGS1)
 AC_SUBST(RSRT_LIBS1)
Index: rsyslog-7.2.7/plugins/imkmsg/kmsg.c
===================================================================
--- rsyslog-7.2.7.orig/plugins/imkmsg/kmsg.c
+++ rsyslog-7.2.7/plugins/imkmsg/kmsg.c
@@ -35,7 +35,7 @@
 #ifdef  OS_LINUX
 #include <sys/klog.h>
 #endif
-#include <json/json.h>
+#include <json.h>
 
 #include "rsyslog.h"
 #include "srUtils.h"
Index: rsyslog-7.2.7/plugins/mmjsonparse/mmjsonparse.c
===================================================================
--- rsyslog-7.2.7.orig/plugins/mmjsonparse/mmjsonparse.c
+++ rsyslog-7.2.7/plugins/mmjsonparse/mmjsonparse.c
@@ -36,7 +36,7 @@
 #include <unistd.h>
 #include <ctype.h>
 #include <libestr.h>
-#include <json/json.h>
+#include <json.h>
 #include "conf.h"
 #include "syslogd-types.h"
 #include "template.h"
Index: rsyslog-7.2.7/plugins/mmnormalize/mmnormalize.c
===================================================================
--- rsyslog-7.2.7.orig/plugins/mmnormalize/mmnormalize.c
+++ rsyslog-7.2.7/plugins/mmnormalize/mmnormalize.c
@@ -40,7 +40,7 @@
 #include <unistd.h>
 #include <libestr.h>
 #include <libee/libee.h>
-#include <json/json.h>
+#include <json.h>
 #include <liblognorm.h>
 #include "conf.h"
 #include "syslogd-types.h"
Index: rsyslog-7.2.7/plugins/ommongodb/ommongodb.c
===================================================================
--- rsyslog-7.2.7.orig/plugins/ommongodb/ommongodb.c
+++ rsyslog-7.2.7/plugins/ommongodb/ommongodb.c
@@ -33,7 +33,7 @@
 #include <stdint.h>
 #include <time.h>
 #include <mongo.h>
-#include <json/json.h>
+#include <json.h>
 /* For struct json_object_iter, should not be necessary in future versions */
 #include <json/json_object_private.h>
 
Index: rsyslog-7.2.7/runtime/msg.c
===================================================================
--- rsyslog-7.2.7.orig/runtime/msg.c
+++ rsyslog-7.2.7/runtime/msg.c
@@ -41,7 +41,7 @@
 #endif
 #include <netdb.h>
 #include <libestr.h>
-#include <json/json.h>
+#include <json.h>
 /* For struct json_object_iter, should not be necessary in future versions */
 #include <json/json_object_private.h>
 #if HAVE_MALLOC_H
Index: rsyslog-7.2.7/runtime/msg.h
===================================================================
--- rsyslog-7.2.7.orig/runtime/msg.h
+++ rsyslog-7.2.7/runtime/msg.h
@@ -30,7 +30,7 @@
 
 #include <pthread.h>
 #include <libestr.h>
-#include <json/json.h>
+#include <json.h>
 #include "obj.h"
 #include "syslogd-types.h"
 #include "template.h"
Index: rsyslog-7.2.7/template.c
===================================================================
--- rsyslog-7.2.7.orig/template.c
+++ rsyslog-7.2.7/template.c
@@ -34,7 +34,7 @@
 #include <string.h>
 #include <ctype.h>
 #include <assert.h>
-#include <json/json.h>
+#include <json.h>
 #include "stringbuf.h"
 #include "syslogd-types.h"
 #include "template.h"
Index: rsyslog-7.2.7/template.h
===================================================================
--- rsyslog-7.2.7.orig/template.h
+++ rsyslog-7.2.7/template.h
@@ -30,7 +30,7 @@
 #ifndef	TEMPLATE_H_INCLUDED
 #define	TEMPLATE_H_INCLUDED 1
 
-#include <json/json.h>
+#include <json.h>
 #include <libestr.h>
 #include "regexp.h"
 #include "stringbuf.h"
